File: ai-stock-advisor/frontend/types/advisor.ts
export interface Recommendation {

    advisor_score: number;

    rating: string;

    recommendation: string;

    confidence: string;

}

export interface Confidence {

    confidence_score: number;

    confidence: string;

    engine_scores: number[];

}

export interface Risk {

    risk_score: number;

    risk_level: string;

}

export interface WeightingEngine {

    score: number;

    weight: number;

    weighted_score: number;

}

export interface Weighting {

    fundamental: WeightingEngine;

    technical: WeightingEngine;

    news: WeightingEngine;

    advisor_score: number;

}

export interface AdvisorResponse {

    engine: string;

    version: string;

    generated_at: string;

    ticker: string;

    advisor_score: number;

    summary: string;

    recommendation: Recommendation;

    confidence: Confidence;

    risk: Risk;

    weighting: Weighting;

    engines: any;

}

File: ai-stock-advisor/backend/models/advisor.py
from pydantic import BaseModel
from typing import Any, Dict


class RecommendationModel(BaseModel):

    advisor_score: float

    rating: str

    recommendation: str

    confidence: str


class ConfidenceModel(BaseModel):

    confidence_score: float

    confidence: str

    engine_scores: list[int]


class RiskModel(BaseModel):

    risk_score: float

    risk_level: str


class AdvisorResponse(BaseModel):

    engine: str

    version: str

    generated_at: str

    ticker: str

    advisor_score: float

    recommendation: RecommendationModel

    confidence: ConfidenceModel

    risk: RiskModel

    weighting: Dict[str, Any]

    engines: Dict[str, Any]